src/main/java/io/jenkins/plugins/coverage/CoverageChecksPublisher.java

```diff
@@ -0,0 +1,104 @@
+package io.jenkins.plugins.coverage;
+
+import hudson.model.Run;
+import io.jenkins.plugins.checks.api.*;
+import io.jenkins.plugins.checks.api.ChecksOutput.ChecksOutputBuilder;
+import io.jenkins.plugins.checks.api.ChecksDetails.ChecksDetailsBuilder;
+import io.jenkins.plugins.coverage.targets.*;
+
+import java.util.*;
+
+/**
+ * Publishes coverage as checks to scm platforms.
+ *
+ * @author Kezhi Xiong
+ */
+class CoverageChecksPublisher {
+    private final CoverageAction action;
+
+    CoverageChecksPublisher(final CoverageAction action) {
+        this.action = action;
+    }
+
+    void publishChecks() {
+        ChecksPublisher publisher = ChecksPublisherFactory.fromRun(action.getOwner());
+        publisher.publish(extractChecksDetails());
+    }
+
+    private ChecksDetails extractChecksDetails() {
+        CoverageResult result = action.getResult();
+        ChecksOutputBuilder outputBuilder = new ChecksOutputBuilder()
+                .withTitle("Code Coverage")
+                .withSummary("## Generated " + result.getResults().size() + " types of coverage.")
+                .withText(extractChecksText(result)); // TODO: improve summary
+
+        return new ChecksDetailsBuilder()
+                .withName("Code Coverage")
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(ChecksConclusion.SUCCESS)
+                .withDetailsURL(action.getOwner().getAbsoluteUrl() + "coverage/")
+                .withOutput(outputBuilder.build())
+                .build();
+    }
+
+    private String extractChecksText(CoverageResult result) {
+        Map<CoverageElement, Ratio> ratios = result.getResults();
+        Map<String, Float> lastRatios = getLastRatios(result);
+
+        StringBuilder text = new StringBuilder();
+        for (Map.Entry<CoverageElement, Ratio> singleRatio : ratios.entrySet()) {
+            text.append("## ")
+                    .append(singleRatio.getKey().getName())
+                    .append(": \n* Coverage: ")
+                    .append(singleRatio.getValue())
+                    .append(" :white_check_mark:");
```

Comment:
consider putting this at the start of the line so that the emoji's are all aligned